File: AD-HOC/1547/1547.java
import java.util.Scanner;

public class Main {
	
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int qtdTestes = scan.nextInt();
		int ganhadorId = 0;
		
		while(qtdTestes-- > 0) {
			int menorDiferenca = 101;
			int qtdAlunos = scan.nextInt();
			int numeroSorteado = scan.nextInt();
			
			for(int i=0; i<qtdAlunos; i++) {
				int diferenca = Math.abs(scan.nextInt() - numeroSorteado);
				
				if(diferenca < menorDiferenca) {
					menorDiferenca = diferenca;
					ganhadorId = i+1;
				}
			}
			System.out.println(ganhadorId);
		}
		scan.close();
	}
}

File: AD-HOC/1547/1547.cpp
#include <iostream>

using namespace std;

int abs(int val) {
	if(val < 0) return val*-1;
	return val;
}

int main () {
	
	int qtdTestes, ganhadorId=0;
	
	scanf("%d", &qtdTestes);
	
	while(qtdTestes--) {
		int qtdAlunos, numeroSorteado, valor, diferenca, menorDiferenca = 101;
		
		scanf("%d%d", &qtdAlunos, &numeroSorteado);
		
		for(int i=0; i<qtdAlunos; i++) {
			scanf("%d", &valor);
			diferenca = abs(valor - numeroSorteado);
			
			if(diferenca < menorDiferenca) {
				menorDiferenca = diferenca;
				ganhadorId = i+1;
			}
		}
		printf("%d\n", ganhadorId);
	}
	return 0;
}
